### PR TITLE
(#195) Allow overriding federation as cli argument and playbook func args

### PR DESCRIPTION
**Pull request by vjanelle**
Adds federation parameters, WIP

Also updates the versions of things

### CODE REVIEW

File: bin/mco [hunk 1]
```diff
@@ -1,64 +1,27 @@
 #!/usr/bin/env ruby
+# frozen_string_literal: true
 
-# For security reasons, ensure that '.' is not on the load path
-# This is primarily for 1.8.7 since 1.9.2+ doesn't put '.' on the load path
-$LOAD_PATH.delete '.'
+#
```
Comment (ripienaar) [inline bin/mco:4]: guessing changes here are a mistake, else i need to look closer but I am not really ready to touch this stuff unless neccesary

File: lib/mcollective/rpc/client.rb [hunk 2]
```diff
@@ -44,6 +44,7 @@ def initialize(agent, flags={}) # rubocop:disable Metrics/MethodLength
         @initial_options = initial_options
 
         @config = initial_options[:config]
+        # @config.federations = initial_options[:federations] if initial_options[:federations]
```
Comment (ripienaar) [inline lib/mcollective/rpc/client.rb:47]: not needed?

**Review by ripienaar — CHANGES_REQUESTED** [review]
Doesnt seem so bad :)

File: bin/mco (hunk 1 above)
Comment (vjanelle) [reply bin/mco:4]: This was auto-generated, but I can remove this from the PR

File: lib/mcollective/rpc/client.rb (hunk 2 above)
Comment (vjanelle) [reply lib/mcollective/rpc/client.rb:47]: probably not, but I haven't finished all of the testing yet

**Pull request merged**